src/databricks/labs/lsql/backends.py

```diff
@@ -277,8 +286,7 @@ def save_table(
         mode: Literal["append", "overwrite"] = "append",
     ) -> None:
         rows = self._filter_none_rows(rows, klass)
-
-        if len(rows) == 0:
+        if not rows and mode == "append":
```

Comment:
Is the `mode == "append"` needed?

Reply:
Yes; without this mode='overwrite' could return early without truncating the file.